hypothesis-python/src/hypothesis/strategies/_internal/featureflags.py

```diff
@@ -58,10 +58,16 @@ def __init__(
         # original model.
         #
         # We implement this as a single 8-bit integer and enable features which
-        # score >= that value. In particular when self.__baseline is 0, all
+        # score >= that value. In particular when self.__p_disabled is 0, all
         # features will be enabled. This is so that we shrink in the direction
         # of more features being enabled.
         if self.__data is not None:
+            # this really messes up our deduplication tracking, because all 255
+            # draws are unique. But we more or less have to choose whether something
+            # is enabled on-demand with a prior probability, rather than choosing what
+            # is enabled up front, because the latter results in a very large choice
+            # sequence when there are lots of possibilities.
+            # (a tradeoff might be selecting up front when there are <= 3 options?)
             self.__p_disabled = self.__data.draw_integer(0, 254) / 255
```

Comment:
See comment; not too sure what to do here. Here's one example of this added redundancy:

```python
@given(st.from_regex(re.compile("[ab]*", re.IGNORECASE), fullmatch=True))
def f(v):
    print(v)
f()
```
which prints "A" once on master but ~8 times on this branch.

Reply:
if `self.__p_disabled` is only ever used as the `p=` argument to `draw_bool()`, and is strictly between 0 and 1, do we need to choose it via the choice sequence at all?  vs. `data.random()`, or `0.5` if that's not available?

Reply:
Sort of. `DataTree` thinks it's flaky because the kwargs `p` passed to `draw_boolean` has changed. It's not *actually* flaky because the choice sequence doesn't depend on p (as long as we're careful about p=0.0 and p=1), but `DataTree` doesn't know that.

We could imagine some kind of "ignore this param for flakiness" api, but I'm not sure how that would look.

Reply:
I think ignoring distribution-only arguments in flakiness-checks makes sense; we can still replay perfectly.